pkg/util/kubernetes/kubernetes.go

```diff
@@ -156,9 +156,9 @@ func addTLSConfig(sconfig *natsconf.ServerConfig, cs v1alpha2.ClusterSpec) {
 
 	if cs.TLS.ServerSecret != "" {
 		sconfig.TLS = &natsconf.TLSConfig{
-			CAFile:   constants.ServerCAFilePath,
-			CertFile: constants.ServerCertFilePath,
-			KeyFile:  constants.ServerKeyFilePath,
+			CAFile:   constants.ServerCertsMountPath + "/" + cs.TLS.ServerSecretCAFileName,
```

Comment:
might be good to start using `filepath.Join(constants.ServerCertsMountPath, cs.TLS.ServerSecretCAFileName)` for these